app/watchers/services_watcher.go

```diff
@@ -93,7 +95,15 @@ func StartServiceWatcher(clientset *kubernetes.Clientset, resyncPeriod time.Dura
 
 	svcw.clientset = clientset
 	svcw.broadcaster = utils.NewBroadcaster()
-	lw := cache.NewListWatchFromClient(clientset.Core().RESTClient(), "services", metav1.NamespaceAll, fields.Everything())
+	lw := &cache.ListWatch{
```

Comment:
Had to switch to this since the fake client does not have an implementation for `clientset.Core().RESTClient()`.

Reply:
On a related note, I don't think we should be using global variable for the watchers, but something that can be addressed later.